### DISCUSSION TITLE
[RFC] API gateway of choice

### DISCUSSION THREAD

**Discussion by aldy505** [TokoBapak/.github#8] [Infrastructure Team Posts]
We need just right API gateway for TokoBapak.

It should:
* Be able to expose UDP and TCP connection
* Be lightweight, mustn't consume as much processing power and RAM
* Be able to expose HTTP and gRPC connection

This decision would be coupled with our orchestrator of choice, we'll do it on a separate thread.

cc @ii64 @zasdaym

**Comment by zasdaym** [comment]
https://www.envoyproxy.io/ will cover all use cases, but the configuration format is not so friendly.

**Comment by aldy505** [comment]
I'm thinking about Traefik, just because I'm used to use that one. But I don't know which one is better for this use case.

**Comment by zasdaym** [comment]
Ah, Traefik is far more simpler. +1

**Comment by ii64** [comment]
Trafik +1
